File: src/problems_all/1281_subtract_the_product_and_sum_of_digits_of_an_integer/1281_subtract_the_product_and_sum_of_digits_of_an_integer_v1.py
"""
https://leetcode.com/problems/subtract-the-product-and-sum-of-digits-of-an-integer/description/?envType=study-plan&id=programming-skills-i
https://leetcode.com/problems/subtract-the-product-and-sum-of-digits-of-an-integer/description/

Runtime: 30 ms
Memory: 13.8 MB
"""


class Solution:
    def subtractProductAndSum(self, n: int) -> int:
        total = 0
        product = 1
        digits = map(int, list(str(n)))

        for i in digits:
            total += i
            product *= i

        result = product - total
        return result


def main():
    n = 4421
    result = Solution().subtractProductAndSum(n)
    print(result)


if __name__ == '__main__':
    main()

File: src/problems_all/1281_subtract_the_product_and_sum_of_digits_of_an_integer/1281_subtract_the_product_and_sum_of_digits_of_an_integer_v1.go
/*
https://leetcode.com/problems/subtract-the-product-and-sum-of-digits-of-an-integer/description/?envType=study-plan&id=programming-skills-i
https://leetcode.com/problems/subtract-the-product-and-sum-of-digits-of-an-integer/description/

Runtime: 0 ms
Memory: 1.9 MB
*/

package main

import "fmt"

func main() {
	n := 4421
	result := subtractProductAndSum(n)
	fmt.Println(result)
}

func subtractProductAndSum(n int) int {
	total := 0
	product := 1

	for n > 0 {
		i := n % 10
		total += i
		product *= i
		n /= 10
	}

	result := product - total
	return result
}
